crates/stark-felt/src/lib.rs

```diff
@@ -26,140 +44,201 @@ pub struct FromBytesError;
 
 impl Felt {
     /// [Felt] constant that's equal to 0.
-    pub const ZERO: Self = Self {};
+    pub const ZERO: Self = Self(FieldElement::<Stark252PrimeField>::const_from_raw(
+        UnsignedInteger::from_u64(0),
+    ));
 
     /// [Felt] constant that's equal to 1.
-    pub const ONE: Self = Self {};
+    pub const ONE: Self = Self(FieldElement::<Stark252PrimeField>::from_hex_unchecked("1"));
 
     /// [Felt] constant that's equal to 2.
-    pub const TWO: Self = Self {};
+    pub const TWO: Self = Self(FieldElement::<Stark252PrimeField>::from_hex_unchecked("2"));
 
     /// [Felt] constant that's equal to 3.
-    pub const THREE: Self = Self {};
+    pub const THREE: Self = Self(FieldElement::<Stark252PrimeField>::from_hex_unchecked("3"));
 
     /// Maximum value of [Felt]. Equals to 2^251 + 17 * 2^192.
-    pub const MAX: Self = Self {};
+    pub const MAX: Self = Self(FieldElement::<Stark252PrimeField>::const_from_raw(
+        UnsignedInteger::from_limbs([544, 0, 0, 32]),
+    ));
 
     /// Creates a new [Felt] from its big-endian representation in a [u8] slice.
-    pub const fn from_bytes_be(_bytes: &[u8]) -> Result<Self, FromBytesError> {
-        todo!()
+    /// This is as performant as [from_bytes_le](Felt::from_bytes_le)
+    pub fn from_bytes_be(bytes: &[u8]) -> Result<Self, FromBytesError> {
+        FieldElement::from_bytes_be(bytes)
+            .map(Self)
+            .map_err(|_| FromBytesError)
     }
 
     /// Creates a new [Felt] from its little-endian representation in a [u8] slice.
-    pub const fn from_bytes_le(_bytes: &[u8]) -> Result<Self, FromBytesError> {
-        todo!()
+    /// This is as performant as [from_bytes_be](Felt::from_bytes_be)
+    pub fn from_bytes_le(bytes: &[u8]) -> Result<Self, FromBytesError> {
+        FieldElement::from_bytes_le(bytes)
+            .map(Self)
+            .map_err(|_| FromBytesError)
     }
 
     /// Converts to big-endian byte representation in a [u8] array.
-    pub const fn to_bytes_be(&self) -> [u8; 32] {
-        todo!()
+    /// This is as performant as [to_bytes_le](Felt::to_bytes_le)
+    pub fn to_bytes_be(&self) -> [u8; 32] {
+        self.0.to_bytes_be()
     }
 
     /// Converts to little-endian byte representation in a [u8] array.
-    pub const fn to_bytes_le(&self) -> [u8; 32] {
-        todo!()
+    /// This is as performant as [to_bytes_be](Felt::to_bytes_be)
+    pub fn to_bytes_le(&self) -> [u8; 32] {
+        self.0.to_bytes_le()
     }
 
     /// Converts to big-endian bit representation.
-    pub const fn to_bits_be(&self) -> BitArray<BitArrayStore> {
-        todo!()
+    /// This is as performant as [to_bits_le](Felt::to_bits_le)
+    pub fn to_bits_be(&self) -> BitArray<BitArrayStore> {
+        let mut limbs = self.0.representative().limbs;
```

Comment:
@LucasLvy what do you think about this?

Reply:
If it changes it should be updated but IDK if it could change so much except if you do a big refacto on the felt type ? @fmoletta

Reply:
IMO even if it changes the worst that can happen is that downstream applications see a performance drop if they bump dependency and didn't notice (or just didn't lock dependencies). They still do compile. They would notice if they really care about performance anyways.

If we really want to, we can always make a semver-incompatible release so that Cargo doesn't auto upgrade.